tools/download-schemas.ts

```diff
@@ -0,0 +1,114 @@
+  self:string;
+  git: string;
+  html: string;
+}
+
+/**
+ * Represents the type of a GitHub content item
+ */
+enum GithubContentItemType {
+  File = 'file',
+  Dir = 'dir'
+}
+
+/**
+ * Represents a GitHub content item
+ */
+interface GithubContentItem {
+  name: string;
+  path: string;
+  sha: string;
+  size: number;
+  url: string;
+  html_url: string;
+  git_url: string;
+  download_url: string;
+  type: GithubContentItemType;
+  _links: GithubContentItemLinks;
+}
+
+/** Resets the destination directory */
+const reset = async (destDir: string) => rimrafP(destDir)
+  .then(() => mkdir(destDir, { recursive: true }))
+  .then(() => writeFile(path.resolve(destDir, 'README.md'), `# Auto generated notice
+This directory and its content has been generated automatically. Do not modify its content, it WILL be lost.`));
+
+/**
+ * A promise that list the schemas in the registry
+ * @param {string} url The registry entry point
+ * @returns {GithubContentItem[]} The list of items
+ */
+const listFiles = async (url: string): Promise<GithubContentItem[]> => fetch(url)
+  .then((res: any) => {
+    if (res.status !== 200) throw `Cannot fetch ${url}`;
+    return res.json();
+  })
+  .then((items: GithubContentItem[]) => {
+    const queue = items.map(i => {
+      if (i.type === GithubContentItemType.Dir) return listFiles(i.url);
+      return Promise.resolve([i]);
+    });
+    return Promise.all(queue);
+  })
+  .then((items: GithubContentItem[][]) => items.reduce((acc, i) => [...acc, ...i]));
+/**
+ * A promise that maps the item paths and their urls
+ * @param {GithubContentItem[]} items The list of items
+ * @returns {Map<string, string>} The path/url map
+ */
+const mapFilePaths = async (items: GithubContentItem[]): Promise<Map<string, string>> => {
+  const filesMap = new Map<string, string>();
+  items.forEach(i => {
+    filesMap.set(
+      i.path,
+      i.download_url
+    );
+  })
+  return Promise.resolve(filesMap);
+};
+/**
+ * A promise to download the file at the provided url to the provided dest
+ * @param {string} url The URL to get the file from
+ * @param {string} dest The destination path
+ * @returns {void}
+ */
+const downloadFile = async (url: string, dest: string): Promise<void> => mkdir(
+    path.resolve(process.cwd(), 'src/lib', dest.split('/').slice(0, -1).join('/')), 
+    { recursive: true }
+  )
+  .then(() => fetch(url))
+  .then((res: any) => res.arrayBuffer())
+  .then((data) => writeFile(path.resolve(process.cwd(), 'src/lib', dest), Buffer.from(data)))
+  ;
+/**
+ * A promise to download the provided files
+ * @param {Map<string, string>} filesMap The path/url map to download to/from
+ * @returns {void}
+ */
+const downloadFiles = async (filesMap: Map<string, string>): Promise<void[]> => Promise.all(Array.from(filesMap).map(([dest, url]) => downloadFile(url, dest)));
+
+/** The schema registry base url */
+const registryUrl = 'https://api.github.com/repos/serverlessworkflow/specification/contents/schema?ref=main';
```

Comment:
we are targeting here to the main branch, witch is "under development".

As it is really easy to regenerate everything now, tks @JBBianchi , once this is merged I am thinking to do the same for https://api.github.com/repos/serverlessworkflow/specification/contents/schema?ref=0.6.x so we can tag a version from v0.6 spec

Reply:
Do you want me to change the current version to 0.6.x before merging? It takes a second. We could also try to match the spec major/minor version wih the package version (the patch would be different though). WDYT?

Reply:
@tsurdilo ^

Reply:
ok @tsurdilo , I am going to create the `severlessworkflow` account in https://www.npmjs.com/,  still have to figure out how to do it... 🙄

Reply:
I think it's as easy as filling the form of [the signup page](https://www.npmjs.com/signup) and then using `npm login` and `npm publish`. Versionning is usually managed in the CI/CD. I know [semantic-release](https://github.com/semantic-release/semantic-release) is a popular tools but I never used it myself.

Maybe you'd want to try to publish on a private GitHub registry to see if the CI/CD works as expected, if the package can actually be used as we expect ... This would avoid to "pollute" npm with broken versions.

Reply:
I think it's as easy as filling the form of [the signup page](https://www.npmjs.com/signup) and then using `npm login` and `npm publish`. Versionning is usually managed in the CI/CD. I know [semantic-release](https://github.com/semantic-release/semantic-release) is a popular tools but I never used it myself.

Maybe you'd want to try to publish on a private GitHub registry to see if the CI/CD works as expected, if the package can actually be used as we expect ... This would avoid to "pollute" npm with broken versions.

Reply:
👍

Reply:
Committed the new version of the downloader (and few adaptations that needed to be done for the whole thing to still work). By default, it will fetch the package.json version and use major/minor for the github ref (so package 0.6.789 will download v0.6). In addition, the downloader also takes a --registry flag to force a provided url (might be usefull for dev or CI/CD): `npm run tools:download-schemas -- --registry=https://api.github.com/repos/serverlessworkflow/specification/contents/schema?ref=main` for instance.